elfi/methods/parameter_inference.py

```diff
@@ -936,54 +935,38 @@ def _report_batch(self, batch_index, params, distances):
             str += "{}{} at {}\n".format(fill, distances[i].item(), params[i])
         logger.debug(str)
 
-    def plot_state(self, **options):
-        """Plot the GP surface
-        
-        This feature is still experimental and currently supports only 2D cases.
-        """
-
-        f = plt.gcf()
-        if len(f.axes) < 2:
-            f, _ = plt.subplots(1,2, figsize=(13,6), sharex='row', sharey='row')
+    def plot_acq_points(self):
+        """Plots the acquisition points.
+            - All pair-wise combinations of the parameters are plotted
+            in the upper triangular partition of the subplots.
+            - The marginals of a single parameter are plotted in the diagonal.
 
+            Note: The method is experimental.
+        """
         gp = self.target_model
+        points_acq = gp.X
+        names_param = self.parameter_names
+        n_params = gp.X.shape[1]
 
-        # Draw the GP surface
-        visin.draw_contour(gp.predict_mean,
-                           gp.bounds,
-                           self.parameter_names,
-                           title='GP target surface',
-                           points = gp.X,
-                           axes=f.axes[0], **options)
+        if n_params == 1:
+            print('Use plot_state() to visualise acquisition points in 1D.')
+        else:
+            vis.plot_acq_points(points_acq, names_param, n_params)
```

Comment:
This can be shortened to `vis.plot_acq_points(points_acq, self.parameter_names)`